test/common/router/router_test.cc

```diff
@@ -258,7 +260,9 @@ class RouterTestBase : public testing::Test {
   std::string upstream_zone_{"to_az"};
   envoy::config::core::v3::Locality upstream_locality_;
   NiceMock<Stats::MockIsolatedStatsStore> stats_store_;
+  Stats::TestUtil::TestStore stats_;
   NiceMock<Upstream::MockClusterManager> cm_;
+  Stats::TestUtil::TestStore cm_stats_;
```

Comment:
Can't you just directly change the mocks to use the new utility class and then you get that everywhere and you can avoid changes in this file entirely?

Reply:
The underlying stats object is not a mock. It's a directly instantiated IsolatedStoreImpl from

So this solution wraps that existing instance.

Reply:
Right, but why can't you just put the wrapper directly into the mock cluster and access it via the same variables that it was before? I'm not sure why there have to be changes to this test.

Reply:
I think you are right: I think that's better, but it does pull in a bit more work into this PR, to allow the mocks to depend on that new class. It's better tho.

Working on it.

Reply:
I think this was an improvement, thanks for the great comments.

I do think we need the string-map though. With fastbuild, and without stirng-maps, test/extensions/filters/network/redis_proxy/command_splitter_impl_test.cc explodes in duration and causes timeouts. This is because there are hundreds of tests due to parameterization. But it turns out the maps are not hard now that I am overriding IsolatedStoreImpl.